improver/cli/generate_clearsky_solar_radiation.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2021 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to run GenerateClearSkySolarRadiation ancillary generation."""
+from datetime import datetime, timezone
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    target_grid: cli.inputcube,
+    time,
+    accumulation_period: int,
+    *,
+    temporal_spacing=30,
+    altitude: cli.inputcube = 0.0,
```

Comment:
These default values are floats rather than `inputcube`s, which makes the type a bit messy and will likely end up needing type-dependent logic inside the processing plugin to handle it. One option would be to have these as optional inputcubes (eg. default value of None) and if it's not provided the constant default value gets used - see orography in [temp_lapse_rate](https://github.com/metoppv/improver/blob/e46866908b599d8aecd674e9d03c156560829155/improver/cli/temp_lapse_rate.py#L43) for an example of this in another CLI.

I'm not sure of the best approach here, will think about this further.

Reply:
Yeah. Clize doesn't seem to allow multiple types, but does seem to allow for setting a default value that differs from the input type as is done here.

The intention here is to allow computation when one or neither of these cubes are passed in, whereby a constant value is used in their place. Are you suggesting that in the case of `None` a cube with constant values is created (possibly from `target_grid`) or assign float values as is done here?

The plan is that within the process function (to come in follow-up PR), once the altitude and/or linke_turbidity cubes have been confirmed to have consistent spatial grid with the target_grid, the data passed to subsequent functions is the underlying numpy array subsequent functions able to handle `ndarrays` or `floats` for these values interchangably.

Reply:
The best approach I can think of so far is:
- the CLI process function declaration has a default value of `None`
- inside the CLI process function, add `if altitude is None` and `if linke_turbidity is None` blocks which construct appropriate cubes based on `target_grid` (filled with the default values)
- the `GenerateClearskySolarRadiation` class works with cube inputs only, so any other direct use of that class would need to construct appropriate cubes

The upside of this is a simple interface with minimal variation in types being passed. The downside is no way to set a non-default constant value for altitude and linke turbidity - any more advanced usage than the default means needing to provide cubes (or NetCDF files).
There are tradeoffs here and I'm open to other opinions.

Reply:
Thanks @tjtg. I've gone with the suggestion above and have added a function to create a constant cube on the same spatial grid as the target grid. If you're happy with this approach, I'll add the associated unit test and hopefully after that we should be close.

Reply:
I've updated this to now use `improver.metadata.utilities.create_new_diagnostic_cube`.